addons/l10n_in_pos/static/src/overrides/components/pos_receipt.xml

```diff
@@ -26,5 +26,29 @@
                 </div>
             </t>
         </xpath>
+        <xpath expr="//div[@class='before-footer']" position="after">
+            <br/>
+            <table t-if="props.data?.l10n_in_hsn_summary and props.data.headerData.company.country_id?.code === 'IN'" style="width:100%;">
+                <tr>
+                    <th class="text-center fw-bolder" colspan="6">--- HSN Summary ---</th>
+                </tr>
+                <tr>
+                    <th class="text-center">HSN Code</th>
+                    <th class="text-center">Rate%</th>
+                    <th class="text-center">CGST</th>
+                    <th class="text-center">SGST</th>
+                    <th class="text-center">IGST</th>
+                    <th class="text-center">CESS</th>
+                </tr>
+                <tr t-foreach="props.data.l10n_in_hsn_summary" t-as="line" t-key="line_index">
+                    <td class="text-center"><t t-out="line_value.l10n_in_hsn_code"/></td>
+                    <td class="text-center"><t t-out="line_value.gst_rate"/> %</td>
+                    <td class="text-center"><t t-out="line_value.CGST.amount.toFixed(2)"/></td>
+                    <td class="text-center"><t t-out="line_value.SGST.amount.toFixed(2)"/></td>
+                    <td class="text-center"><t t-out="line_value.IGST.amount.toFixed(2)"/></td>
+                    <td class="text-center"><t t-out="line_value.CESS.amount.toFixed(2)"/></td>
```

Comment:
instead of `amount.toFixed(2)`, you should probably use `env.utils.formatCurrency(line_value.CGST.amount, false)`.

In fact, it's better if you pass here the formatted value.